src/pepr/prometheus/index.ts

```diff
@@ -0,0 +1,60 @@
+import { Capability, K8s, Log, kind } from "pepr";
+import { Prometheus } from "../operator/crd";
+
+export const prometheus = new Capability({
+  name: "prometheus",
+  description: "UDS Core Capability for the Prometheus stack.",
+});
+
+const { When } = prometheus;
+
+/**
+ * Mutate a service monitor to enable mTLS metrics
+ *
+ */
+When(Prometheus.ServiceMonitor)
+  .IsCreatedOrUpdated()
+  .Mutate(async sm => {
+    // Provide an opt-out of mutation to handle complicated scenarios
+    if (sm.Raw.metadata?.labels?.["uds/skip-sm-mutate"]) {
+      return;
+    }
+    const namespaces = sm.Raw.spec?.namespaceSelector?.matchNames || [sm.Raw.metadata?.namespace];
+    let istioInjected = false;
+
+    for (const ns of namespaces) {
+      if (ns === undefined) {
+        return;
+      }
+      const namespace = await K8s(kind.Namespace).Get(ns);
+      if (
+        namespace.metadata?.labels &&
+        namespace.metadata.labels["istio-injection"] === "enabled"
+      ) {
+        istioInjected = true;
+      }
```

Comment:
I think this could be pulled out into a helper function like `isIstioInjected`.  I believe you are after if any of the namespaces (or namespace of the SM) are istio injected.